### ISSUE TITLE
Group Visibility Issue 

### ISSUE THREAD

**Issue by s4069b** [tbar0970/jethro-pmm#390]
I created a couple of groups and set as 'visible to me only'. 

However if I log-in as another user I can see the groups under 'groups/list-all' and if I click on the group I can see the members of the group.

**Comment by tbar0970** [comment]
Thanks for reporting. Just to confirm- is this on a new install?

**Comment by s4069b** [comment]
Not 100% sure what you mean by that clarifying question. 
This is a 'stock' install of version v2.19.2 
And a bit more info ... may not be relevant?
- I have had to tell the server to use php 5.?
- and required MDB2 as well

I have been testing with it for a few weeks and it has a few hundred people and about 50 different groups. 

Does that help?

**Comment by tbar0970** [comment]
Your system hasn't been upgraded from an older version - that answers my question :)

**Comment by s4069b** [comment]
Oh. Ok. glad that helps. 

But it gives me another question tho. Do I need to upgrade?  under admin-upgrade so I read " Your system is up to date on version v2.19.2"

**Comment by tbar0970** [comment]
No need to upgrade. It just helps me figure out where the problem comes from. I'll fix it in the next release.

**Comment by s4069b** [comment]
OK thanks.

**Comment by tbar0970** [comment]
There's a fix coming in v2.21, but in the meantime you can fix it with this SQL:

```
DROP VIEW person_group;
CREATE VIEW person_group AS
SELECT * from _person_group g
WHERE
  getCurrentUserID() IS NOT NULL
  AND
  ((g.owner IS NULL) OR (g.owner = getCurrentUserID()))
  AND
  (NOT EXISTS (SELECT * FROM account_group_restriction gr WHERE gr.personid  = getCurrentUserID())
  OR g.id IN (SELECT groupid FROM account_group_restriction gr WHERE gr.personid = getCurrentUserID()));
```

**Comment by s4069b** [comment]
thank you
